torchtitan/components/validate.py

```diff
@@ -126,9 +130,7 @@ def validate(
         else:
             global_avg_loss = loss
```

Comment:
I think here you need the following to send loss from GPU back to CPU before you can even print them out. This is the branch when no FSDP/CP is used (i.e. during TP / PP / single-device training)
```suggestion
            global_avg_loss = loss.item()
```